### ISSUE TITLE
No tests for tagged releases

### ISSUE THREAD

**Issue by wenkokke** [agda/agda#7164]
The tagged releases seem to delete the `agda-tests` entry from `Agda.cabal`.
I suppose this is so that people running `cabal install Agda` don't get an `agda-tests` executable on their system.
However, this makes it essentially impossible to run the tests for a tagged release, e.g., to validate the executable you've built.

Can we move the `agda-tests` entry to its own `.cabal` file and include that as a package in `cabal.project`?

**Comment by andreasabel** [comment]
Yeah, stripping _all_ tests is something that I have had my doubts in.  We should at least ship some basic test suites that do elementary validation of the core functionality (e.g. can process a sample agda file that contains a tour through the features).

I have done some work in 2021 towards running the testsuite also under Windows:
- https://github.com/agda/agda/issues/5549

However, I didn't really think we should ship our whole testsuite that takes 2 hours to run...

Isolating a small shippable testsuite would be the path I'd take here.

**Comment by wenkokke** [comment]
Two hours?!

*realizes she's never run the test suite*

**Comment by wenkokke** [comment]
My current approach to test executables is to simply check the standard library, which I suppose achieves the goal of making sure the executable works... but I was hoping for something that covers a few more odd edge cases.

**Comment by andreasabel** [comment]
(@UlfNorell says: installing Agda is a minimal test of its functionality, because it invokes itself on the primitive modules.)

I am anyway for a small `cabal test` with a tour through the features...

**Comment by wenkokke** [comment]
> (@UlfNorell says: installing Agda is a minimal test of its functionality, because it invokes itself on the primitive modules.)
> 
> 
> 
> I am anyway for a small `cabal test` with a tour through the features...
> 
> 

That's really only true if I'm building Agda from source, not if I'm running `pip install agda`.

**Comment by andreasabel** [comment]
Another advantage to have a separate package for the tests would be that we do not need the manual step to remove the test-suite before a release.

If we make the tests into their own package, we probably would prefix all the modules with `Test.Agda.` and call the package `agda-tests`, as its executable would be named.

**Comment by plt-amy** [comment]
Doesn't most of the test suite runtime come from stdlib and cubical? For validating an Agda executable it would suffice to run just `succeed` and `fail`, right? And those are pretty fast, even though there's a lot — I would be shocked if they took more than 10m to run (even 5 would be slightly surprising!).

**Comment by andreasabel** [comment]
According to https://github.com/agda/agda/actions/runs/13160036811 the tests that do not involve the std-lib or cubical still add up to 15 min.
Note that there are the `latex` tests which actually invoke `pdflatex`.  And there are compiler tests that invoke GHC.  

But I guess this is not an obstacle to making the tests into a separate package.  No one is obliged to actually run the tests... :-D

**Comment by wenkokke** [comment]
There's not really any reason not to package them, since nobody is obliged to run them... and even if they want to run them, they presumably can be run in groups?

**Comment by andreasabel** [comment]
Agda dev 2025-02-19: Make `agda-tests` into its own package by simply adding a `agda-tests.cabal` to `test/` and add a `cabal.project`.  This would not make `agda-tests` shippable yet (as we would also need to add the actual tests to `data-files:` and change the logic where the test runners look for these files).
